mitodl/ocw-studio
# Umar/4499 false warnings of missing metadata and has unpublished changes

### What are the relevant tickets?
closes https://github.com/mitodl/hq/issues/4499

### Description (What does it do?)
One time migration to include the missing hide_download value in metadata and update default values of unpublished changes to False

### How can this be tested?
#### Pre-Testing steps
1. Create a new course.
2. Go to `admin` panel and find the related `Website` and `WebsiteContent (type: website)` objects
3. In the `Website` object, check that the `has_unpublsihed_live` and `has_unpublished_draft` checkboxes should be checked already.
4. Check the `metadata` field in `WebsiteContent (type: website)` object. if it has `hide_download` field, remove it for testing.
    - (Optional) If you are unable to find the `WebsiteContent (type: website)` object, follow the steps below to create it.
        - Go to the Studio and search for the added course.
        - Open the metadata form and save it after filling the required fields.
        - `docker-compose exec -it web ./mange.py shell`
        - `from content_sync.tasks import update_websites_in_root_website`
        - `update_websites_in_root_website()`
5. Go to the Studio and search for the added course. Open the publish drawer. It should have the `You have unpublished changes.` warning.

#### Testing steps
1. Switch to `umar/4499-false-warnings-of-missing-metadata-and-has-unpublished-changes` branch in `ocw-studio` and restart containers.
2. Run migrations: `docker-compose exec web ./manage.py migrate websites`
3. Create a new course again and locate the related `Website` and `Website Content (type: website)` objects
4. In the `Website` object, check that the `has_unpublsihed_live` and `has_unpublished_draft` checkboxes should be unchecked by default.
5. Check the `metadata` field in `WebsiteContent (type: website)` object of the course created before migration. `hide_download` field should be present.
6. Go to the Studio and search for the added course. Open the publish drawer. The `You have unpublished changes.` warning should be removed.
